packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -853,8 +890,26 @@ class _RenderSegmentedControl<T extends Object> extends RenderBox
   }
 
   // Intrinsic Dimensions
+  double get separatorWidth => _kSeparatorInset.horizontal + _kSeparatorWidth;
+  double get totalSeparatorWidth => separatorWidth * (childCount ~/ 2);
 
-  double get totalSeparatorWidth => (_kSeparatorInset.horizontal + _kSeparatorWidth) * (childCount ~/ 2);
+  int getClosestSegmentIndex(double dx) {
+    int index = 0;
+    RenderBox? child = firstChild;
+    while (child != null) {
+      final _SegmentedControlContainerBoxParentData childParentData = child.parentData! as _SegmentedControlContainerBoxParentData;
+      final double clampX = clampDouble(dx, childParentData.offset.dx, child.size.width + childParentData.offset.dx);
+
+      if (dx <= clampX) {
+        break;
+      }
+
+      index++;
```

Comment:
Then this method could return an OOB index (`length`) no?

Reply:
oh sorry, I missed dragging gesture may go out of bounds. Added a unit test to make sure no out of range exception happen.